extension/src/XliffFunctions.ts

```diff
@@ -299,68 +295,62 @@ export function refreshSelectedXlfFileFromGXlf(
           }
           refreshResult.numberOfUpdatedNotes++;
         }
-        if (langTransUnit.sourceIsEmpty() && langTransUnit.targetIsEmpty()) {
+        if (
+          langTransUnit.sourceIsEmpty() &&
+          langTransUnit.targetIsEmpty() &&
+          lfSettings.preferLockedTranslations
+        ) {
           langTransUnit.insertCustomNote(
             CustomNoteType.refreshXlfHint,
             RefreshXlfHint.emptySource
           );
-          refreshResult.numberOfReviewsAdded++;
+          langTransUnit.target.translationToken = TranslationToken.review;
         }
         formatTransUnitForTranslationMode(
-          languageFunctionsSettings.translationMode,
+          lfSettings.translationMode,
           langTransUnit
         );
-        detectInvalidValues(langTransUnit, languageFunctionsSettings);
+        detectInvalidValues(langTransUnit, lfSettings);
         if (langTransUnit.needsReview(true)) {
           refreshResult.numberOfReviewsAdded++;
         }
       }
       newLangXliff.transunit.push(langTransUnit);
       langXliff.transunit.splice(langXliff.transunit.indexOf(langTransUnit), 1); // Remove all handled TransUnits -> The rest will be deleted.
-    } else {
+    } else if (!sortOnly) {
       // Does not exist in target
```

Comment:
This comment just tricked me, so it might be time to change it a bit... :)

Is this better?
```suggestion
      // TransUnit does not exist in language xlf
```